for-demos/serve_hostname/serve_hostname.go

```diff
@@ -88,8 +90,18 @@ func main() {
 			log.Printf("HTTP request from %s", r.RemoteAddr)
 			fmt.Fprintf(w, "%s", hostname)
 		})
-		go log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
+		go func() {
+			// Run in a closure so http.ListenAndServe doesn't block
```

Comment:
Ah you need to evaluate the arguments to execute the fatalf and the closure explicitly requires lazy evaluation.